arangod/RocksDBEngine/RocksDBSettingsManager.cpp

```diff
@@ -80,14 +84,11 @@ arangodb::Result writeCounterValue(
   rocksdb::Status s =
       rtrx->Put(RocksDBColumnFamily::definitions(), key.string(), value);
   if (!s.ok()) {
-    rtrx->Rollback();
-    auto rStatus = convertStatus(s);
-    LOG_TOPIC(WARN, Logger::ENGINES)
-        << "writing counters failed: " << rStatus.errorMessage();
-    return rStatus;
+    return std::make_pair(convertStatus(s), returnSeq);
```

Comment:
Is it correct to return `returnSeq = baseSeq` here? Don't we need to keep `pair.second._sequenceNum` around still?

Reply:
no thats the point, otherwise we never make any progress

Reply:
Isn't this equivalent to ignoring the error and pretending we wrote out the value? Can't we end up with incorrect counters this way?

Reply:
the code in `RocksDBSettingsManager::sync` will abort the entire operation, or am I wrong ?

Reply:
Ah, yes, you're right. So essentially the return seq value doesn't actually matter here.